tools/nix/flake.nix

```diff
@@ -11,44 +11,49 @@
     # Also see the 'stable-packages' overlay at 'overlays/default.nix'.
 
     flake-utils.url = "github:numtide/flake-utils";
+
+    devenv.url = "github:cachix/devenv";
   };
 
   outputs = {
     nixpkgs,
     flake-utils,
+    devenv,
     ...
-  }:
-    flake-utils.lib.eachDefaultSystem
+  }@inputs:
     # Creates an attribute map `{ devShells.<system>.default = ...}`
-    # by calling this function:
-    (
-      system: let
+    flake-utils.lib.eachDefaultSystem (system:
+      let
         # Import nixpkgs and load it into pkgs.
         pkgs = import nixpkgs {
           inherit system;
         };
 
-        # Things needed at build-time.
-        packagesBasic = with pkgs; [
+        # python environment defined separately
+        pythonModules = import ./python-uv.nix {
```

Comment:
I would call that file rather python-modules or so, because it can be mistaken that it returns a python derivation